xcube_zenodo/preload.py

```diff
@@ -0,0 +1,299 @@
+# The MIT License (MIT)
+# Copyright (c) 2024 by the xcube development team and contributors
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import logging
+import os
+import shutil
+import threading
+import time
+from typing import Callable, Union
+import tarfile
+import zipfile
+
+import IPython
+import IPython.display
+import numpy as np
+import tabulate
+import requests
+import xarray as xr
+from xcube.core.store import DataStoreError
+from xcube.core.store import MutableDataStore
+
+from ._utils import estimate_file_format
+from ._utils import translate_data_id2uri
+
+LOG = logging.getLogger(__name__)
+
+
+class Event:
```

Comment:
A better name could be `Task` instead. Could you explain why you chose `Event` and maybe then we could decide if it makes sense to rename it.